### PR TITLE
PIM-961: PIM | Export | Template Export Should Dynamically Establish Data Row Containing Attributes

### PR DESCRIPTION
**Pull request by KimJinPang**
https://github.com/PropelPLM/pim-data-service/assets/77341283/1c676d8f-82ba-4da4-8c7e-d811af823147
*Note a slight change after recording - removed the <Invalid Attribute> tag to standardize with xlsx

- added support for multiple header rows
- some refactoring to reduce time complexity
- fixed tests (`templateHeaders` is a list of lists in order to support multiple headers so length of `templateHeaders` in test file is 1 instead of 2)
- added edge case support for invalid attributes

### CODE REVIEW

**Review by mfullmore — APPROVED** [review]

File: legacy/PimRecordListHelper.js [hunk 1]
```diff
@@ -717,27 +721,45 @@ async function addExportColumns(
           field = field.slice(11, -1);
           exportColumns.push({
             fieldName: defaultColumns.get(field),
-            label: templateHeaders[i],
+            label: templateHeaders[lastHeaderRowIndex][i],
```
Comment (clawyq) [inline legacy/PimRecordListHelper.js:724]: do we need to make sure `templateHeaders[lastHeaderRowIndex]` is never null

Comment (KimJinPang) [reply legacy/PimRecordListHelper.js:724]: Because `lastHeaderRowIndex = templateHeaders.length - 1` we wont get array index out of bounds

**Review by clawyq — APPROVED** [review]
